frontend/src/Components/UI/Button.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { ButtonHTMLAttributes } from 'react'
+
+import classNames from 'classnames'
+
+import styles from './Button.module.css'
+
+/**
+ * Interface for Button component props
+ * @interface ButtonProps
+ * @extends {ButtonHTMLAttributes<HTMLButtonElement>}
+ */
+interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
+  /** Button variant/style type
+   * @default 'default'
+   */
+  variant: 'default' | 'primary' | 'link' | 'danger'
```

Comment:
please check BUTTON_TYPES in ConfirmDialog

these should be consolidated (also I think the implementation for styles mapping in ConfirmDialog is better, as it's in a single place) 

perhaps in a separate PR